src/ray/raylet/lineage_cache.cc

```diff
@@ -211,7 +211,12 @@ void LineageCache::AddReadyTask(const Task &task) {
   }
 }
 
-uint64_t LineageCache::CountUnsubscribedLineage(const TaskID &task_id) const {
+uint64_t LineageCache::CountUnsubscribedLineage(const TaskID &task_id,
+                                                std::unordered_set<TaskID> &seen) const {
```

Comment:
Think you can maybe default initialize here so you don't have to pass it in on the root call `const std::unordered_set<TaskID> &seen = std::unordered_set<TaskID>()` (but is potentially uglier).

Reply:
I think I prefer the current way.